--- a/chrome/browser/ui/tab_helpers.cc
+++ b/chrome/browser/ui/tab_helpers.cc
@@ -30,1 +30,2 @@
 #include "chrome/browser/ui/tab_contents/core_tab_helper.h"
+#include "chromium_src/overlay/chrome/browser/multi_session/fingerprint_seed_delivery.h"
@@ -120,2 +121,5 @@ void TabHelpers::AttachTabHelpers(content::WebContents* web_contents) {
   CoreTabHelper::CreateForWebContents(web_contents);
+
+  // CHROMELEON_HOOK: Attach FingerprintSeedDelivery to handle seed passing.
+  multi_session::FingerprintSeedDelivery::CreateForWebContents(web_contents);
 }
